### PR TITLE
Add a column is_obsolete to system.settings and similar tables

### LINKED ISSUE #50819: Add a column `is_obsolete` to `system.settings` and similar tables.
**Use case**

```
Warnings:
 * Some obsolete setting is changed. Check 'select * from system.settings where changed' and read the changelog.
```

but how do I know?

**Additional context**

The warning message should list the specific obsolete settings as well.

### PR DESCRIPTION
**Pull request by ucasfl**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add a column is_obsolete to system.settings and similar tables. Closes #50819.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
This is an automated comment for commit 4a7761c16210c7e2eccc0b26e172ec8dc7e6c183 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/50826/4a7761c16210c7e2eccc0b26e172ec8dc7e6c183/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Docs Check</td><td>Builds and tests the documentation</td><td>🟡 pending</td></tr>
</table>

**Review by alexey-milovidov — APPROVED** [review]

**Comment by ucasfl** [comment]
I don't known why some tests failed.

**Comment by alexey-milovidov** [comment]
We have to fix these tests before merging.

File: src/Interpreters/Context.cpp [hunk 1]
```diff
@@ -781,8 +781,10 @@ Strings Context::getWarnings() const
     {
         if (setting.isValueChanged() && setting.isObsolete())
         {
-            common_warnings.emplace_back("Some obsolete setting is changed. "
-                                         "Check 'select * from system.settings where changed' and read the changelog.");
+            common_warnings.emplace_back(
+                "Obsolete setting `" + setting.getName()
+                + "` is changed. "
+                  "Check 'select * from system.settings where changed' and read the changelog.");
```
Comment (tavplubix) [inline src/Interpreters/Context.cpp:787]:
```suggestion
                  "Check 'select * from system.settings where changed and is_obsolete' and read the changelog.");
```

File: src/Interpreters/Context.cpp [hunk 2]
```diff
@@ -781,8 +781,10 @@ Strings Context::getWarnings() const
     {
         if (setting.isValueChanged() && setting.isObsolete())
         {
-            common_warnings.emplace_back("Some obsolete setting is changed. "
-                                         "Check 'select * from system.settings where changed' and read the changelog.");
+            common_warnings.emplace_back(
+                "Obsolete setting `" + setting.getName()
```
Comment (tavplubix) [inline src/Interpreters/Context.cpp:785]: It will produce multiple warnings, one for each changed obsolete setting. Is it expected?

Comment (tavplubix) [reply src/Interpreters/Context.cpp:785]: I mean, maybe we can print one message with the list of such settings

**Comment by alexey-milovidov** [comment]
It suggest adding/modifying the documentation.

**Comment by ucasfl** [comment]
> @ucasfl, It says that the documentation has to be updated.

I don't find the places need to update in document.

**Comment by alexey-milovidov** [comment]
Here: https://clickhouse.com/docs/en/operations/system-tables/settings

**Comment by ucasfl** [comment]
> Here: https://clickhouse.com/docs/en/operations/system-tables/settings

Updated.

**Review by alexey-milovidov — APPROVED** [review]

**Review by alexey-milovidov — APPROVED** [review]

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**